s3transfer/__init__.py

```diff
@@ -260,10 +260,11 @@ def _calculate_file_size(self, fileobj, requested_size, start_byte,
         return min(max_chunk_size, requested_size)
 
     def read(self, amount=None):
+        amount_left = max(self._size - self._amount_read, 0)
```

Comment:
I'd prefer we just not update this implementation. This is a legacy implementation that is not used in boto3 nor the CLI and likely will go away in the future. So I don't think we should be making updates to it to indicate we are even trying to maintain this.